internal/controller/api_to_config.go

```diff
@@ -514,6 +514,18 @@ func localPrefPrefixesToMap(withLocalPref []v1beta1.LocalPrefPrefixes) (localPre
 		localPrefForPrefixV6: map[string]uint32{},
 	}
 
+	// error when input is
+	// Prefixes:  []string{"192.0.2.0/24"},LocalPref: 100},
+	// Prefixes:  []string{"100.0.0.0/24"},LocalPref: 100}
+	// we enforce : []string{"192.0.2.0/24","100.0.0.0/24"},LocalPref: 100}
+	seen := make(map[uint32]bool)
+	for _, e := range withLocalPref {
+		if _, exists := seen[e.LocalPref]; exists {
+			return localPrefPrefixes{}, fmt.Errorf("duplicate localPref")
```

Comment:
let's give a better error here, so it's clear what are the offending elements. Something along the line of "multiple  entries with the same localPref: 100"

Reply:
Also also, can this be part of the outer loop below? So it's clear that we are addressing different local prefs in each row and we can focus on not having duplicate prefixes: if each row relates to a different local pref, then the only thing we care about in the inner loop is not to have duplicate prefixes